app/models/tube.rb

```diff
@@ -20,9 +20,21 @@ class Tube < ApplicationRecord
   scope :by_barcode, ->(*barcodes) { where(barcode: barcodes) }
   scope :by_pipeline,
         lambda { |pipeline|
-          joins(:container_materials).where(
-            'container_materials.material_type LIKE ?', "#{pipeline.capitalize}::%"
-          )
+          case pipeline
```

Comment:
I wonder if there is a better way? It might point to a bit of an issue with our design.

Reply:
Because we are doing OO there should be an option using either composition or polymorphism or even metaprogramming. This is definitely much better than the old code and is a move in the right direction. I don't think this is something we need to fix now but is worth further discussion.

Reply:
Yeah this SQL took me a while to figure out but its not very elegant. I had to make this change as a side effect of some other work but I didnt want to spend too long on it. I am not sure what the performance will be like down the line so it would be good to come back to this and see if we can improve on it.